ext/crypto/import_key.rs

```diff
@@ -685,12 +793,81 @@ fn import_key_ec(
           }
         }
       };
-      ImportKeyResult::Ec {
+      Ok(ImportKeyResult::Ec {
         raw_data: RawKeyData::Public(data),
-      }
+      })
     }
-    _ => return Err(unsupported_format()),
-  })
+    KeyData::Pkcs8(data) => {
+      match named_curve {
+        EcNamedCurve::P256 => {
+          let secret_key = p256::SecretKey::from_pkcs8_der(&data).unwrap();
+
+          let point =
+            secret_key.public_key().as_affine().to_encoded_point(false);
+
+          // 3.
+          if point.is_identity() {
+            return Err(data_error("Invalid key data"));
+          }
+        }
+        EcNamedCurve::P384 => {
+          //@todo(sean-wykes) Validate P384 secret-key on import(pkcs8)
```

Comment:
This needs to be done before landing.

Reply:
Missing ProjectiveArithmetic in p384 crate. For now, have changed code to throw an "Unsupported curve" error, instead of blindly importing the PKCS8.

Reply:
:+1: